### PR TITLE
Fixes #287: Docs: Improved Mock chapter

### LINKED ISSUE #287: Update of mock section
Update of mock section, based on review.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.
Ready for review and merge.

Note: The Python script was tested, but requires pywbem with its PR https://github.com/pywbem/pywbem/pull/1821

### CODE REVIEW

File: docs/mock_support.rst [hunk 1]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
-  insert objects into the repository.
-
-  It may also implement pywbem CIM method callbacks, (:meth:`pywbem_mock.method_callback_interface`)
-  for handling InvokeMethod operations requested from pywbemcli.
-
-Pywbemcli logging (`` -l`` or ``--log``) can be enabled when the
-`--mock-server` option is enabled. However, since the mock support does not use
-HTTP, the `http`  of the log option will generate no output.
+Pywbemcli has support for mocking a WBEM server with the general option
+``--mock-server`` (or ``-m``). This allows executing pywbemcli against a mock
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: DONE: Changed to "option `--mock-server/-m`".

File: docs/mock_support.rst [hunk 2]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
+WBEM server that is automatically created under the covers in pywbemcli,
+rather than a real WBEM server.
+
+The ``--mock-server`` option is mutually exclusive with the ``--server`` and
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: This is the general introduction. I think it is important to understand that they are mutually exclusive. If I were interested in mock support, I would start reading the section about it, and not the option description.

File: docs/mock_support.rst [hunk 3]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
+    mock repository.
+
+  * ``SERVER`` (:class:`pywbem.WBEMServer`):
+    This object is layered on top of the ``CONN`` object and provides access
+    to higher level features of the mock WBEM server, such as getting the
+    Interop namespace, adding namespaces, or building more complex objects
+    for the mock repository.
+
+  * ``VERBOSE`` (bool):
+    A flag that represents the use of the ``--verbose`` general option of
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: Ok. DONE as suggested.

File: docs/mock_support.rst [hunk 4]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
-
-  It may also implement pywbem CIM method callbacks, (:meth:`pywbem_mock.method_callback_interface`)
-  for handling InvokeMethod operations requested from pywbemcli.
-
-Pywbemcli logging (`` -l`` or ``--log``) can be enabled when the
-`--mock-server` option is enabled. However, since the mock support does not use
-HTTP, the `http`  of the log option will generate no output.
+Pywbemcli has support for mocking a WBEM server with the general option
+``--mock-server`` (or ``-m``). This allows executing pywbemcli against a mock
+WBEM server that is automatically created under the covers in pywbemcli,
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: It is key to understand that nothing else is needed but the pywbemcli command. No separate mock server. We can discuss how we get that across, though.

File: docs/mock_support.rst [hunk 5]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
+The ``--mock-server`` option is mutually exclusive with the ``--server`` and
+``--name`` options, since each defines a WBEM server.
+
+The automatically created mock WBEM server has a in-memory repository for
+CIM objects (qualifier declarations, classes, and instances) and supports
+CIM namespaces. The operations performed against the mock WBEM server cause
+that mock repository to be inspected or manipulated accordingly.
+
+The mock repository can be loaded with CIM objects from files specified as
+an argument to the ``--mock-server`` option. Each use of the option specifies
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: I don't know either. We should pick one and use that consistently.

File: docs/mock_support.rst [hunk 6]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
+
+  The MOF file may define CIM qualifier declarations, CIM classes and CIM
+  instances.
+
+  At this point, these CIM objects can be added to only one
+  :term:`CIM namespace` in the repository of the mock WBEM server, namely the
+  default namespace of the connection (see ``--default-namespace`` global
+  option).
+
+  If a CIM object already exists in the repository, it is being updated
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: DONE

File: docs/mock_support.rst [hunk 7]
```diff
@@ -8,78 +8,91 @@ Mock WBEM server support
...
+  :class:`~pywbem.CIMInstance` for representing CIM objects, and add them to
+  the mock repository via calls to
+  :meth:`pywbem_mock.FakedWBEMConnection.add_cimobjects`.
+
+  The Python script can also extend the capabilities of the mock WBEM server
+  by implementing callbacks via :func:`pywbem_mock.method_callback_interface`,
+  for handling CIM method invocations against the mock WBEM server.
+
+Pywbemcli logging (``-l`` or ``--log`` general option) can be used together
+with the mock support. However, since the mock support does not use HTTP(S),
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: DONE: Dropped.

**Comment by andy-maier** [comment]
Updated the commit with some resolutions and marked those comments as resolved.

The unresolved comments are up for discussion.

**Comment by KSchopmeyer** [comment]
SUGGESTION: Add two lines to the python example.

if VERBOSE:
    CONN.display_repository()

File: docs/mock_support.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply docs/mock_support.rst]: Karl agrees leave as is

File: docs/mock_support.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply docs/mock_support.rst]: drop under the covers.

File: docs/mock_support.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply docs/mock_support.rst]: We will use the word argument in place of value.  Karl will review the general options also

File: docs/mock_support.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/mock_support.rst]: DONE

**Pull request merged**